### DISCUSSION TITLE
Are there any conditions for chunks to be split when using the multi entry point option?

### DISCUSSION THREAD

**Discussion by Dailyscat** [webpack/webpack#14647] [Q&A]
i'm using webpack4 in vue-cli

The directory structure of my project is like the picture below.

[directory][1]

Bundles corresponding to each entry point, plus chunk-common and chunk-vendors, are created.

But I have a question here when bundles are created at each entry point.
 Assuming you are using the components in src/components/,
Applied so that the component can be referenced like `webpack:///./src/views/sample/components/CategoryTreeSample.vue?646f` in the bundle created in src/views/sample/main.js and the chunk-commons bundle it's done
However, assuming that a component named a is used under the directory `src/views/sample/component/` at two different entry points
This component a is not included in each bundle, but only in chunk-common.

I understand by looking at the example that the common components used in different entry points are included in chunk-common.
https://github.com/webpack/webpack/tree/main/examples/multiple-entry-points
https://github.com/webpack/webpack/tree/main/examples/common-chunk-and-vendor-chunk

I wonder why the components under `src/components/` are included in chunk-common and also included in the bundle of entries used.

The component under `src/views/sample/components/` contains information only in chunk-common.

I think it would be more appreciated if you add the webpack code part as a reference in this regard.
Thanks for the reply in advance!!

my webpack config
Since it is a config added from the vue cli, I think you only need to check the entry and context.

    {
	"mode": "production",
	"context": "/Users/user/Documents/admin_non_upgrade/front/src/main/vue",
	"devtool": "source-map",
	"node": {
		"setImmediate": false,
		"dgram": "empty",
		"fs": "empty",
		"net": "empty",
		"tls": "empty",
		"child_process": "empty"
	},
	"output": {
		"path": "/Users/user/Documents/admin_non_upgrade/front/src/main/resources/static",
		"filename": "js/[name].js",
		"publicPath": "/",
		"chunkFilename": "js/[name].js"
	},
	"resolve": {
		"alias": {
			"@": "/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src",
			"vue$": "vue/dist/vue.runtime.esm.js"
		},
		"extensions": [
			".mjs",
			".js",
			".jsx",
			".vue",
			".json",
			".wasm"
		],
		"modules": [
			"node_modules",
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/node_modules",
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/node_modules/@vue/cli-service/node_modules"
		]
	},
	"resolveLoader": {
		"modules": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/node_modules/@vue/cli-plugin-eslint/node_modules",
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/node_modules/@vue/cli-plugin-babel/node_modules",
			"node_modules",
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/node_modules",
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/node_modules/@vue/cli-service/node_modules"
		]
	},
	"optimization": {
		"splitChunks": {
			"cacheGroups": {
				"vendors": {
					"name": "chunk-vendors",
					"test": {},
					"priority": -10,
					"chunks": "all"
				},
				"common": {
					"name": "chunk-common",
					"minChunks": 2,
					"priority": -20,
					"chunks": "initial",
					"reuseExistingChunk": true
				},
				"vendor": {
					"test": {},
					"chunks": "initial",
					"name": "chunk-xlsx",
					"priority": 10,
					"enforce": true
				}
			}
		},
		"chunkIds": "named"
	},
	"plugins": [
		{},
		{
			"definitions": {
				"process.env": {
					"NODE_ENV": "\"production\"",
					"BASE_URL": "\"/\""
				}
			}
		},
		{
			"options": {},
			"logger": {},
			"pathCache": {},
			"fsOperations": 0,
			"primed": false
		},
		{
			"compilationSuccessInfo": {},
			"shouldClearConsole": true,
			"formatters": [
				null,
				null,
				null,
				null
			],
			"transformers": [
				null,
				null,
				null,
				null
			],
			"previousEndTimes": {}
		},
		{
			"options": {
				"filename": "css/[name].css",
				"ignoreOrder": false,
				"allChunks": true,
				"chunkFilename": "css/[name].css"
			}
		},
		{
			"options": {
				"sourceMap": false,
				"cssnanoOptions": {
					"preset": [
						"default",
						{
							"mergeLonghand": false,
							"cssDeclarationSorter": false
						}
					]
				}
			}
		},
		{
			"options": {
				"context": null,
				"hashFunction": "md4",
				"hashDigest": "hex",
				"hashDigestLength": 4
			}
		},
		{},
		{
			"options": {
				"resourceRegExp": {},
				"contextRegExp": {}
			}
		}
	],
	"entry": {
		"sample": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/sample/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"modelProduct": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/modelProduct/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"productCategory": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/productCategory/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"productCategoryKpi": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/productCategory/kpi/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"productCategoryDetail": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/productCategory/detail/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"productCategoryMapping": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/productCategoryMapping/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"searchEvaluation": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/searchEvaluation/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"modal": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/modal/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"merchantFeedInfo": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/merchants/feedInfo/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"topCycle": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/topCycle/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"products": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/products/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"prohibitionKeyword": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/keyword/prohibition/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"coordinates": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/coordinates/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"productChangeHistory": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/productChangeHistory/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"merchantManagement": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/merchants/management/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"cmpManagement": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/keyword/cmpManagement/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"categoryAffiliate": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/category/affiliate/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"categoryManagement": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/category/management/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"categoryStandardSearch": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/category/standardSearch/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"catalogChangeHistory": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/catalogChangeHistory/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"allowedProduct": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/keyword/allowedProduct/main.js",
			"dependencies/lcp-common-window-dialog"
		],
		"zProductDetail": [
			"/Users/user/Documents/admin_non_upgrade/front/src/main/vue/src/views/products/details/main.js",
			"dependencies/lcp-common-window-dialog"
		]
	},
	"performance": {
		"maxAssetSize": 1000000,
		"hints": false
	}
	

    

  [1]: https://i.stack.imgur.com/BaNdY.png

**Comment by alexander-akait** [comment]
I think better place to ask vue-cli developers for this

**Reply by Dailyscat** [reply]
@alexander-akait The vue cli only provides the result of compiling through webpack with that setting.
So, the location of the context and the location of the entry point seem to play a role in making the chunk, can you help?

**Reply by alexander-akait** [reply]
Yes, they play role, I can't answer on questions because I don't have a source code, also you provide examples of webpack v5, but you are on webpack v4, there are a lot of breaking changes

**Reply by Dailyscat** [reply]
Oh yes, thanks for the reply, I'll check it out
